src/components/Home/Home.js

```diff
@@ -32,34 +36,130 @@ function Home(props) {
     function handleSignupClick() {
       setIsSignupPopupOpen(true);
       setIsSigninPopupOpen(false);
-  }
+    }
 
-  function handleSignupSubmit() {
-    setIsSignupSuccessOpen(true);
-    setIsSignupPopupOpen(false);
-  }
+    function handleSignupSuccess() {
+      setIsSignupSuccessOpen(true);
+      setIsSignupPopupOpen(false);
+    }
       
     function closeAllPopups() {
       setIsSigninPopupOpen(false);
       setIsSignupPopupOpen(false);
       setIsSignupSuccessOpen(false);
     }
+
+    // const results = document.querySelector(".search-results");
+    // const preloader = document.querySelector(".preloader");
+    // const notFound = document.querySelector(".result-not-found");
+    // preloader.setAttribute("style", "display: block;");
+    // results.setAttribute("style", "display: none;");
+
+
+    function handleSearchSubmit(e) {
+      e.preventDefault();
+      const preloader = document.querySelector(".preloader");
+      const results = document.querySelector(".search-results");
+      const notFound = document.querySelector(".result-not-found");
+      const searchInput = document.querySelector(".search__form-input").value;
+      preloader.setAttribute("style", "display: block;");
+      notFound.setAttribute("style", "display: none;");
+      newsapi.getNews(searchInput)
+      .then((res) => {
+        preloader.setAttribute("style", "display: none;");
+        results.setAttribute("style", "display: flex;");
```

Comment:
`("style", "display: flex;")` you need to add/remove `css`-classes in react to achieve this logics. You shouldn't add styles directly to `DOM`-elements. Please, correct it everywhere in your project